lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -429,9 +420,11 @@ public function diffColumn(Column $column1, Column $column2)
         $changedProperties = [];
 
         foreach (['type', 'notnull', 'unsigned', 'autoincrement'] as $property) {
-            if ($properties1[$property] != $properties2[$property]) {
-                $changedProperties[] = $property;
+            if ($properties1[$property] === $properties2[$property]) {
```

Comment:
This looks risky given the boolean types of the above may be not enforced. But I'd rather fix the root cause if the issue exists than the comparison itself.

Reply:
#3427